File: app/src/main/java/com/xomena/cmpfutboltfe/MyPlace.java
package com.xomena.cmpfutboltfe;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import com.google.android.gms.maps.model.LatLng;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

public class MyPlace {
    private static final String LOG_TAG = "MyPlace";

    private String placeID;
    private LatLng latLng;
    private String mName;
    private String address;

    public MyPlace(Set<String> place) {
        Iterator<String> it = place.iterator();

        try {
            String val = it.next();
            String[] data = val.split("###");

            placeID = data[0];
            latLng = new LatLng(Double.valueOf(data[1]), Double.valueOf(data[2]));
            mName = data[3];
            address = data[4];
        } catch (Exception e) {
            Log.e(LOG_TAG, "Bad place data", e);
        }
    }

    public String getName() {
        return mName;
    }

    public String getPlaceID() {
        return placeID;
    }

    public LatLng getLatLng() {
        return latLng;
    }

    public String getAddress() {
        return address;
    }

    public static List<MyPlace> createPlacesList(Context context) {
        List<MyPlace> places = new ArrayList<>();

        SharedPreferences sharedPref = context.getSharedPreferences(
                MainActivity.PLACES_SHARED_PREF, Context.MODE_PRIVATE);
        Set<String> keys = sharedPref.getStringSet(ManagePlacesActivity.STORED_KEYS, new LinkedHashSet<String>());

        for (String key : keys) {
            Set<String> pdata = sharedPref.getStringSet(key, new LinkedHashSet<String>());
            if (pdata.size() == 1) {
                places.add(new MyPlace(pdata));
            }
        }

        return places;
    }
}

File: app/src/main/java/com/xomena/cmpfutboltfe/model/MyPlace.kt
package com.xomena.cmpfutboltfe.model

import android.content.Context
import android.util.Log
import com.google.android.gms.maps.model.LatLng
import com.xomena.cmpfutboltfe.MainActivity
import com.xomena.cmpfutboltfe.ManagePlacesActivity

data class MyPlace(
    val placeID: String,
    val latLng: LatLng,
    val name: String,
    val address: String
) {
    companion object {
        private const val LOG_TAG = "MyPlace"

        @JvmStatic
        fun fromStringSet(place: Set<String>): MyPlace? {
            return try {
                val data = place.first().split("###")
                MyPlace(
                    placeID = data[0],
                    latLng = LatLng(data[1].toDouble(), data[2].toDouble()),
                    name = data[3],
                    address = data[4]
                )
            } catch (e: Exception) {
                Log.e(LOG_TAG, "Bad place data", e)
                null
            }
        }

        @JvmStatic
        fun createPlacesList(context: Context): List<MyPlace> {
            val sharedPref = context.getSharedPreferences(
                MainActivity.PLACES_SHARED_PREF,
                Context.MODE_PRIVATE
            )
            val keys = sharedPref.getStringSet(
                ManagePlacesActivity.STORED_KEYS,
                linkedSetOf()
            ) ?: emptySet()

            return keys.mapNotNull { key ->
                val pdata = sharedPref.getStringSet(key, linkedSetOf()) ?: emptySet()
                if (pdata.size == 1) {
                    fromStringSet(pdata)
                } else {
                    null
                }
            }
        }
    }
}
